codes.go

```diff
@@ -204,864 +756,388 @@ const (
 
-var KEYName = map[EvCode]string{
-	0:   "KEY_RESERVED",
-	1:   "KEY_ESC",
-	2:   "KEY_1",
-	3:   "KEY_2",
-	4:   "KEY_3",
-	5:   "KEY_4",
-	6:   "KEY_5",
-	7:   "KEY_6",
-	8:   "KEY_7",
-	9:   "KEY_8",
-	10:  "KEY_9",
-	11:  "KEY_0",
-	12:  "KEY_MINUS",
-	13:  "KEY_EQUAL",
-	14:  "KEY_BACKSPACE",
-	15:  "KEY_TAB",
-	16:  "KEY_Q",
-	17:  "KEY_W",
-	18:  "KEY_E",
-	19:  "KEY_R",
-	20:  "KEY_T",
-	21:  "KEY_Y",
-	22:  "KEY_U",
-	23:  "KEY_I",
-	24:  "KEY_O",
-	25:  "KEY_P",
-	26:  "KEY_LEFTBRACE",
-	27:  "KEY_RIGHTBRACE",
-	28:  "KEY_ENTER",
-	29:  "KEY_LEFTCTRL",
-	30:  "KEY_A",
-	31:  "KEY_S",
-	32:  "KEY_D",
-	33:  "KEY_F",
-	34:  "KEY_G",
-	35:  "KEY_H",
-	36:  "KEY_J",
-	37:  "KEY_K",
-	38:  "KEY_L",
-	39:  "KEY_SEMICOLON",
-	40:  "KEY_APOSTROPHE",
-	41:  "KEY_GRAVE",
-	42:  "KEY_LEFTSHIFT",
-	43:  "KEY_BACKSLASH",
-	44:  "KEY_Z",
-	45:  "KEY_X",
-	46:  "KEY_C",
-	47:  "KEY_V",
-	48:  "KEY_B",
-	49:  "KEY_N",
-	50:  "KEY_M",
-	51:  "KEY_COMMA",
-	52:  "KEY_DOT",
-	53:  "KEY_SLASH",
-	54:  "KEY_RIGHTSHIFT",
-	55:  "KEY_KPASTERISK",
-	56:  "KEY_LEFTALT",
-	57:  "KEY_SPACE",
-	58:  "KEY_CAPSLOCK",
-	59:  "KEY_F1",
-	60:  "KEY_F2",
-	61:  "KEY_F3",
-	62:  "KEY_F4",
-	63:  "KEY_F5",
-	64:  "KEY_F6",
-	65:  "KEY_F7",
-	66:  "KEY_F8",
-	67:  "KEY_F9",
-	68:  "KEY_F10",
-	69:  "KEY_NUMLOCK",
-	70:  "KEY_SCROLLLOCK",
-	71:  "KEY_KP7",
-	72:  "KEY_KP8",
-	73:  "KEY_KP9",
-	74:  "KEY_KPMINUS",
-	75:  "KEY_KP4",
-	76:  "KEY_KP5",
-	77:  "KEY_KP6",
-	78:  "KEY_KPPLUS",
-	79:  "KEY_KP1",
-	80:  "KEY_KP2",
-	81:  "KEY_KP3",
-	82:  "KEY_KP0",
-	83:  "KEY_KPDOT",
+// Values describing the status of a force-feedback effect
+const (
+	FF_STATUS_STOPPED = 0x00
+	FF_STATUS_PLAYING = 0x01
+	FF_STATUS_MAX     = 0x01
+
+	// Force feedback effect types
+
+	FF_RUMBLE   = 0x50
+	FF_PERIODIC = 0x51
+	FF_CONSTANT = 0x52
+	FF_SPRING   = 0x53
+	FF_FRICTION = 0x54
+	FF_DAMPER   = 0x55
+	FF_INERTIA  = 0x56
+	FF_RAMP     = 0x57
+
+	FF_EFFECT_MIN = FF_RUMBLE
+	FF_EFFECT_MAX = FF_RAMP
+
+	// Force feedback periodic effect types
+
+	FF_SQUARE   = 0x58
+	FF_TRIANGLE = 0x59
+	FF_SINE     = 0x5a
+	FF_SAW_UP   = 0x5b
+	FF_SAW_DOWN = 0x5c
+	FF_CUSTOM   = 0x5d
+
+	FF_WAVEFORM_MIN = FF_SQUARE
+	FF_WAVEFORM_MAX = FF_CUSTOM
+
+	// Set ff device properties
+
+	FF_GAIN       = 0x60
+	FF_AUTOCENTER = 0x61
+
+	// ff->playback(effect_id = FF_GAIN) is the first effect_id to
+	// cause a collision with another ff method, in this case ff->set_gain().
+	// Therefore the greatest safe value for effect_id is FF_GAIN - 1,
+	// and thus the total number of effects should never exceed FF_GAIN.
+
+	FF_MAX_EFFECTS = FF_GAIN
+
+	FF_MAX = 0x7f
+	FF_CNT = (FF_MAX + 1)
+)
+
+// Type to String
+
+var INPUTToString = map[EvProp]string{
+	0x0: "INPUT_PROP_POINTER",
+	0x1: "INPUT_PROP_DIRECT",
+	0x2: "INPUT_PROP_BUTTONPAD",
+	0x3: "INPUT_PROP_SEMI_MT",
+	0x4: "INPUT_PROP_TOPBUTTONPAD",
+	0x5: "INPUT_PROP_POINTING_STICK",
+	0x6: "INPUT_PROP_ACCELEROMETER",
+
+	0x1f: "INPUT_PROP_MAX",
+	0x20: "INPUT_PROP_CNT",
```

Comment:
Key values could be referenced by corresponding constants here.  
Also duplicated key values could be leaved commented instead of skipping them entirely.